docs/contributor/03-11-hap-rules.md

```diff
@@ -0,0 +1,249 @@
+
+## Uniqnuess & Priority
+
+Only one rule can be triggered. If more than one rule entry match the request then only one is selected and applied. The process of selecting the best matching rule is based on a rule uniqueness and priority.
+Rule entry uniqueness is determined by its plan and input parameters with its values (identification attributes) not including input parameters with `*`. 
+Output parameters are not taken into account when establishing rule entry uniqueness. For example the following rule will fail on startup because every SKR can be matched by both rules:
+
+```
+hap:
+  rule: 
+    - gcp 
+    - gcp -> S                      # invalid entry, output attributes does not take part into uniqnuess check
+    - gcp(PR=*)                     # invalid entry, both can be applied to the same skr
+    - gcp(HR=europe-west3)          # valid entry, new HR attribute makes the rule unique
+    - gcp(PR=*, HR=europe-west3)    # invalid rules, the same as previous one because of addition of `PR=*` attribute
+```
+
+Rule configuration must contain only unique entries in the scope of that list.
+If rule configuration contains duplicated rule entries an error that fails KEB startup is returned.
+
+Rule entry priority is selected by sorting all rule entries that apply to the request by the number of identification attributes they contain. 
+For example, a rule with no attributes and only a plan has lower priority then a rule with the same plan but a platform region attribute (`gcp` < `gcp(PR=cf-sa30)`).
+After sorting the entry that specifies most attributes (hence is the most specific) is selected. 
+Input attributes with value `*` are not taken into account when calculating priority.
+
+The following example shows priority of rules starting from the lowest priority:
+
+```
+aws -> S                              # search labels: hyperscalerType: aws, shared: true
+aws(PR=cf-eu11) -> EU                 # search labels: hyperscalerType: aws_cf-eu11, euAccess: true
+aws(PR=cf-eu11, HR=westeu) -> EU, S   # search labels: hyperscalerType: aws_cf-eu11_westeu, shared: true, euAccess: true
+```
+
+## Validation
+
+KEB validates HAP Rules during startup. If the configuration is invalid, KEB will not start and an error message will be displayed in the logs. 
+
+The constraints used for validation during KEB startup include:
+* Rules format check - all the rules must comply with the format specified above.
+* Every supported plan needs at least one rule entry, if no rule entry is defined for a plan then an error is returned during KEB startup.
+* Uniqueness validation check - KEB checks if all rule entries are unique in the scope of the rule. Specifying more than one entry with the same number of identification attributes is prohibited. If more than one such attribute is specified then the error is returned failing KEB startup. For more details see [Uniqnuess & Priority](#uniqnuess--priority) section. 
```

Comment:
```suggestion
* Uniqueness validation check: KEB checks if all rule entries are unique in the rule's scope. You must not specify more than one entry with the same number of identification attributes. Otherwise, the error is returned, failing KEB startup. For more details, see the [Uniqueness & Priority](#uniqueness-and-priority) section. 
```